evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -923,6 +921,12 @@ impl<F: Field> Transition<F> for Interpreter<F> {
     }
 }
 
+impl<F: Field> StateLog for Interpreter<F> {
+    fn log_log(_: log::Level, _: &str) {}
+
+    fn log_debug(_: &str) {}
+}
```

Comment:
The problem with this is that fully removes interpreter logging. If we need it for testing / debugging (cf comment above on `User Instruction` log removal, then we won't be able to do anything with the logs.

Reply:
Should we call `log::trace!` in both cases? For `log_log` the level is computed dynamically before calling it in transition.rs:282. Should we just ignore the level for the interpreter and call `log::trace!` anyway?

Reply:
If we do then the PR loses its point, because we will log everything during JD analysis though

Reply:
I think if we only allow the interpreter to log instead of the `GenerationState`, then we shouldn't get "double logs", since `jumpdest_analysis` is called with the `GenerationState`.

Reply:
I'm confused. Jumpdest simulation is done within the interpreter no? i.e. we initialize a new `Interpreter` instance between the start and halt labels of JD analysis.

Reply:
> we will log everything during JD analysis though

You mean we will log everything with level == trace?

Reply:
Yes, when trace is enabled (which is the default when debugging failures)

Reply:
so we would need another lower level then?

Reply:
You can't with the `log` crate, that'd require a custom new implementation

Reply:
I guess the simplest solution would be to add the `#[cfg(debug_assertions]` as we were doing  before the unification.

Reply:
Actually, why not just enable Interpreter logs (as in feat/continuations)? In that case, we can also check whether we're in the jumpdest analysis with self.is_jumpdest_analysis.